addons/cert.py

```diff
@@ -234,6 +245,101 @@ def msc30(data):
+                
+
+# STR07-C
+# Use the bounds-checking interfaces for string manipulation
+def str07(data):
+    for token in data.tokenlist:
+        if token.str=='strcpy' or token.str=='strcat' or token.str=='fputs':
+            parent = token.astParent
+            if parent is None:
+                continue
+            if not parent.astParent :
+                reportError(token, 'style', 'Use the bounds-checking interfaces for string manipulation', 'STR07-C')
+
+# STR11-C
+# Do not specify the bound of a character array initialized with a string literal
+def str11(data):
+    for token in data.tokenlist:
+        if token.isString:
+            strlen = token.strlen
+            parent = token.astParent
+            if parent is None:
+                continue
+            parentOp1 = parent.astOperand1
+            if parent.isAssignmentOp and not parentOp1.valueType is None:
+                valueToken = parentOp1.astOperand2
+                if valueToken is None:
+                    continue
+                if valueToken.isNumber and int(valueToken.str)==strlen:
+                    reportError(valueToken, 'style', 'Do not specify the bound of a character array initialized with a string literal', 'STR11-C')
+
+# MSC24-C
+# Do not use deprecated or obsolescent functions
+def msc24(data):
+    for token in data.tokenlist:
+        if token.str == 'asctime':
+            reportError(token,'style','Do no use asctime() better use asctime_s()', 'MSC24-C')
+        elif token.str == 'atof':
+            reportError(token,'style','Do no use atof() better use strtod()', 'MSC24-C')
+        elif token.str == 'atoi':
+            reportError(token,'style','Do no use atoi() better use strtol()', 'MSC24-C')
+        elif token.str == 'atol':
+            reportError(token,'style','Do no use atol() better use strtol()', 'MSC24-C')
+        elif token.str == 'atoll':
+            reportError(token,'style','Do no use atoll() better use strtoll()', 'MSC24-C')
+        elif token.str == 'ctime':
+            reportError(token,'style','Do no use ctime() better use ctime_s()', 'MSC24-C')
+        elif token.str == 'fopen':
+            reportError(token,'style','Do no use fopen() better use fopen_s()', 'MSC24-C')
+        elif token.str == 'freopen':
+            reportError(token,'style','Do no use freopen() better use freopen_s()', 'MSC24-C')
+        elif token.str == 'rewind':
+            reportError(token,'style','Do no use rewind() better use fseek()', 'MSC24-C')
+        elif token.str == 'setbuf':
+            reportError(token,'style','Do no use setbuf() better use setvbuf()', 'MSC24-C')
+
+
+# MSC25-C
+# Do not use insecure or weak cryptographic algorithms
+# this checker will only report known weak algorithms if the well known function names are used.
+def msc25(data):
+    for token in data.tokenlist:
+        if 'md4_init' in token.str.lower() or 'md4_update' in token.str.lower() or 'md4_final' in token.str.lower():
```

Comment:
I feel skeptic about this matching. this will warn about `symptomd4_initype`. If we know that a function is insecure/weak maybe we can configure that in a cfg file and warn in core cppcheck.